README.md
# Podcast RSS Editor
Started a [design podcast](https://anyway.fm/) couple of years ago, and in the very beginning,
I manually updated the RSS feed for every episode and very soon, I got sick of the 'tag dancing',
so tried a little bit building this tool, a very simple Podcast RSS editor in PHP for myself.

![](https://raw.githubusercontent.com/JJYing/Podcast-RSS-Editor/master/assets/screenshot.jpg)

Forked from https://github.com/JJYing/Podcast-RSS-Editor.  Many thanks to JJ Ying!

# 01. Features
- Add / Remove / Duplicate episodes from existing RSS feeds.
- Analyzing publish date and episode duration information.

# 02. Usage
- Copy the `example.xml` to a new file for each feed you want to maintain.
- Edit all the channel information for each file.
- Place the XML file in the `feeds` directory.
- Ensure all the XML files in `feeds` as well as the config.php have write permissions
  by the web server (e.g. chmod 666 config.php feeds/example.xml).
- To add and/or edit episodes, first select the desired feed in the '**Settings**' panel.

# 03. Authentication
 - There's a very simple authentication done with `.htaccess`, so all you gotta do is to generate
the `.htpasswd` file content [in this site](http://www.htaccesstools.com/htpasswd-generator/)
and replace it.
 - The parameter **AuthUserFile** of the `.htaccess` file is the path to the `.htpasswd` that
 **must to be the real path**. Be sure to change it as needed.
 - By default authentication is: user `admin` and password `admin`

# 04. Alternatives
- [Itunes podcast XML generator](http://codepen.io/jon-walstedt/pen/jsIup) by Jon Wålstedt
- [Podcast Generator](http://www.podcastgenerator.net/)
- Any Podcast hosting services.

# 05. Credits
- Some icons used are from the set [Squid Ink Line Icons](http://thesquid.ink/line-icons/)


CHANGELOG.md
# Change Log
- [Oct 18, 2021] - Restructured by [Marco Moreno](https://github.com/CheckeredFlag)
    - Incorporated PR from [Marco Braghim](https://github.com/marcobraghim) for Portuguese translation & security.
    - Moved feeds into new `feeds` directory having public access.
    - Added channel title to top of episode list.
    - Changed CSS color to use a variable for easier theming.
    - Added `view` get parameter to replace negative episode number.
    - Segregated portions of PHP from HMTL.
    - Use server redirects instead of javascript.
    - Implemented sessions for config handling.
    - Improved sanitizing user input.
    - Overhauled config & language loading/saving.
    - Defined many unitialized variables to eliminate notices.
    - Updated to use php 7 features.
    - Added support for [VS Code](https://code.visualstudio.com) and [Lando](https://lando.dev).
    - Moved changes to new `CHANGELOG.md`.
- [Feb 3, 2019] Updated the layout compatibility.
- [Jul 27, 2017] Added ‘**Preview**’ function for episode link, image file and audio file.
- [Dec 13, 2016] Added ‘**duplicate**’ and ‘**delete**’ functions.
- [Jul 20, 2016] Added multi-language support: Simplified Chinese, Traditional Chinese, English.


assets/logout.js
/*
 * This function will try to force browser to request
 * login again, but it's not really confiable, so avoid
 * to depend entirely on it.
 */
function logout() {

    var baseURL = window.location.origin+window.location.pathname;

    var xmlhttp;
    if (window.XMLHttpRequest) {
        xmlhttp = new XMLHttpRequest();
    }
    // code for IE
    else if (window.ActiveXObject) {
        xmlhttp=new ActiveXObject("Microsoft.XMLHTTP");
    }
    if (window.ActiveXObject) {
        // IE clear HTTP Authentication
        document.execCommand("ClearAuthenticationCache");
        window.location.href=baseURL;
    } else {
        xmlhttp.open("GET", baseURL+'assets/logout.js', true, "logout", "logout");
        xmlhttp.send("");
        xmlhttp.onreadystatechange = function() {
            if (xmlhttp.readyState == 4) {window.location.href=baseURL;}
        }
    }
    return false;
}


language/Portugues - BR.php
<?php
return [
    0   => '<strong>Podcast</strong> RSS Editor',
    1   => 'Episódios',
    2   => 'Dashboard',
    3   => 'Configurações',
    4   => 'Dias desde o último Ep.',
    5   => 'Total Episódios',
    6   => 'Add novo Episódio',
    7   => 'Título:',
    8   => 'Publicado em:',
    9   => 'Duração:',
    10  => 'Link:',
    11  => 'Autor:',
    12  => 'Imagem de capa:',
    13  => 'Arquivo de áudio:',
    14  => 'Descrição:',
    15  => 'Dias',
    16  => 'desde o último episódio, e ciclo de atualização médio é de ',
    17  => ' dias.',
    18  => 'Episódios',
    19  => 'no total depois de sua primeira exibição <br />',
    20  => ' anos atrás.',
    21  => 'Horas',
    22  => 'podcasted, que é ',
    23  => ' \'O Hobbit\' trilogia combinada.',
    24  => 'KB',
    25  => 'por arquivo de RSS, incluindo ',
    26  => ' links <br />in shownotes.',
    27  => 'Pessoas',
    28  => 'mostrado na lista de autores.<br /> ',
    29  => ' é o mais popular.',
    30  => 'Dom',
    31  => 'Seg',
    32  => 'Ter',
    33  => 'Qua',
    34  => 'Qui',
    35  => 'Sex',
    36  => 'Sab',
    37  => 'A duração dos últimos 10 episódios vs. média de duração (',
    38  => 'minutos) de todos os tempos.',
    39  => 'Selecione o arquivo para editar:',
    40  => 'E quem sabe trocar o idioma:',
    41  => 'Salvar',
    42  => 'Novo episódio',
    43  => 'Editar Episódio #',
    44  => ' ',
    45  => 'Mês',
    46  => 'Dia',
    47  => 'Ano',
    48  => 'Hora',
    49  => 'Minuto',
    50  => 'Segundo',
    51  => 'HH',
    52  => 'MM',
    53  => 'SS',
    54  => 'Configurações salvas',
    55  => 'Episódio atualizado',
    56  => 'Novo episódio criado',
    57  => 'Você tem certeza de que quer excluir “',
    58  => '”?',
    59  => 'Diga adeus àquele episódio, ele se foi para sempre.',
    60  => 'Duplicar este episódio',
    61  => 'Excluir este episódio',
    62  => 'Dublicado com sucesso',
    63  => 'Episódios',
    64  => 'Complete hosts list: ',
    65  => ' anos atrás',
    66  => ' meses atrás',
    67  => ' semanas atrás',
    68  => ' dias atrás',
    69  => ' horas atrás',
    70  => 'Publicado',
    71  => 'Ontem',
    72  => 'Visualizar',
];


language/Chinese - Simplified.php
<?php
return [
    0   => '<strong>Podcast</strong> RSS Editor',
    1   => '播客音频',
    2   => '统计面板',
    3   => '基础设置',
    4   => '距离上期更新天数',
    5   => '总期数',
    6   => '新加一期',
    7   => '标题：',
    8   => '发布时间：',
    9   => '时长：',
    10  => '链接：',
    11  => '演出者：',
    12  => '封面图 URL：',
    13  => '音频文件 URL：',
    14  => '简介：',
    15  => '天没更新',
    16  => '平均更新周期为 ',
    17  => ' 天',
    18  => '总期数',
    19  => '第一期发布至今约 ',
    20  => ' 年',
    21  => '总时长',
    22  => '等于刷 ',
    23  => ' 遍〈霍比特人〉三部曲',
    24  => 'KB 文件尺寸',
    25  => 'Shownotes 中包含了 ',
    26  => ' 个参考链接',
    27  => '个人',
    28  => '已经播音过，',
    29  => ' 出场最多',
    30  => '日',
    31  => '一',
    32  => '二',
    33  => '三',
    34  => '四',
    35  => '五',
    36  => '六',
    37  => '最近 10 期的节目时长 vs. 开播以来的平均时长（',
    38  => ' 分钟）',
    39  => '选择想要编辑的 RSS 文件',
    40  => '选择界面语言',
    41  => '保存',
    42  => '新节目',
    43  => '编辑第 ',
    44  => ' 期',
    45  => '月',
    46  => '日',
    47  => '年',
    48  => '时',
    49  => '分',
    50  => '秒',
    51  => '小时',
    52  => '分钟',
    53  => '秒',
    54  => '设置已保存',
    55  => '节目已更新',
    56  => '新节目创建完毕',
    57  => '确定要删除「',
    58  => '」吗？',
    59  => '已经删除完毕',
    60  => '复制这期节目',
    61  => '删除这期节目',
    62  => '节目已复制',
    63  => '期节目',
    64  => '完整列表：',
    65  => ' 年前',
    66  => ' 个月前',
    67  => ' 周前',
    68  => ' 天前',
    69  => ' 小時前',
    70  => '刚刚发布',
    71  => '昨天',
    72  => '打开预览',
];


language/Chinese - Traditional.php
<?php
return [
    0   => '<strong>Podcast</strong> RSS Editor',
    1   => '播客音訊',
    2   => '統計面板',
    3   => '基礎設定',
    4   => '距離上期更新天數',
    5   => '總期數',
    6   => '新加一期',
    7   => '標題：',
    8   => '發佈時間：',
    9   => '時長：',
    10  => '連結：',
    11  => '演出者：',
    12  => '封面圖 URL:',
    13  => '音訊檔案 URL:',
    14  => '簡介：',
    15  => '天沒更新',
    16  => '平均更新週期為 ',
    17  => ' 天',
    18  => '總期數',
    19  => '第一期發佈至今約 ',
    20  => ' 年',
    21  => '總時長',
    22  => '等於刷 ',
    23  => ' 遍〈哈比人〉三部曲',
    24  => 'KB 檔案尺寸',
    25  => 'Shownotes 中包含了',
    26  => ' 個連結',
    27  => '個人',
    28  => '出現在節目中，',
    29  => ' 出場最多',
    30  => '日',
    31  => '一',
    32  => '二',
    33  => '三',
    34  => '四',
    35  => '五',
    36  => '六',
    37  => '最近 10 期的節目時長 vs. 開播以來的平均時長（',
    38  => '分鐘）',
    39  => '選擇想要編輯的 RSS 檔案',
    40  => '選擇介面語言',
    41  => '保存',
    42  => '新節目',
    43  => '編輯第 ',
    44  => ' 期',
    45  => '月',
    46  => '日',
    47  => '年',
    48  => '時',
    49  => '分',
    50  => '秒',
    51  => '小時',
    52  => '分鐘',
    53  => '秒',
    54  => '設定已保存',
    55  => '節目已更新',
    56  => '新節目創建完畢',
    57  => '確定要刪除「',
    58  => '」嗎？',
    59  => '已經刪除完畢',
    60  => '複製本期節目',
    61  => '刪除本期節目',
    62  => '節目已複製',
    63  => '期節目',
    64  => '完整清單：',
    65  => ' 年前',
    66  => ' 個月前',
    67  => ' 周前',
    68  => ' 天前',
    69  => ' 小時前',
    70  => '剛剛發佈',
    71  => '昨天',
    72  => '打開預覽',
];


language/English.php
<?php
return [
    0   => '<strong>Podcast</strong> RSS Editor',
    1   => 'Episodes',
    2   => 'Dashboard',
    3   => 'Settings',
    4   => 'Days Since Last Ep.',
    5   => 'Total Episodes',
    6   => 'Add New Episode',
    7   => 'Title:',
    8   => 'Publish Time:',
    9   => 'Duration:',
    10  => 'Link:',
    11  => 'Author:',
    12  => 'Cover Image:',
    13  => 'Audio File:',
    14  => 'Description:',
    15  => 'Days',
    16  => 'since last episode, and average update cycle is ',
    17  => ' days.',
    18  => 'Episodes',
    19  => 'in total after your first show <br />',
    20  => ' years ago.',
    21  => 'Hours',
    22  => 'podcasted, which is ',
    23  => ' \'The Hobbit\' trilogy combined.',
    24  => 'KB',
    25  => 'for RSS file size, including ',
    26  => ' links <br />in shownotes.',
    27  => 'People',
    28  => 'showed in the aurthor list.<br /> ',
    29  => ' is the hottest one.',
    30  => 'Sun',
    31  => 'Mon',
    32  => 'Tue',
    33  => 'Wed',
    34  => 'Thu',
    35  => 'Fri',
    36  => 'Sat',
    37  => 'The durations of last 10 episodes vs. average duration (',
    38  => 'minutes) of all time.',
    39  => 'Select the file to edit:',
    40  => 'And maybe switch Language:',
    41  => 'Save',
    42  => 'New Episode',
    43  => 'Edit Episode #',
    44  => ' ',
    45  => 'Month',
    46  => 'Day',
    47  => 'Year',
    48  => 'Hour',
    49  => 'Minute',
    50  => 'Second',
    51  => 'HH',
    52  => 'MM',
    53  => 'SS',
    54  => 'Settings saved',
    55  => 'Episode updated',
    56  => 'New episode created',
    57  => 'Are you sure you want to delete “',
    58  => '”?',
    59  => 'Say goodbay to that episode, it\'s gone forever.',
    60  => 'Duplicate this episode',
    61  => 'Delete this episode',
    62  => 'Successfully Duplicated',
    63  => 'Episodes',
    64  => 'Complete hosts list: ',
    65  => ' years ago',
    66  => ' months ago',
    67  => ' weeks ago',
    68  => ' day ago',
    69  => ' hours ago',
    70  => 'Just Published',
    71  => 'Yesterday',
    72  => 'Open Preview',
];
